content/program-security.md

```diff
@@ -0,0 +1,825 @@
+# Create a Basic Program, Part 3 - Basic Security and Validation
+
+# Lesson Objectives
+
+*By the end of this lesson, you will be able to:*
+
+- Explain the importance of "thinking like an attacker"
+- Understand basic security practices
+- Perform owner checks
+- Perform signer checks
+- Validate accounts passed in the program
+- Perform basic data validation
+
+# TL;DR
+
+- **Thinking like an attacker** means asking "How do I break this?"
+- Perform **owner checks** to ensure that the provided account is owned by the public key you expect, e.g. ensuring that an account you expect to be a PDA is owned by `program_id`
+- Perform **signer checks** to ensure that any account modification has been signed by the right party or parties
+- **Account validation** entails ensuring that provided accounts are the accounts you expect them to be, e.g. rederiving PDAs with the expected seeds to make sure the address matches the provided account
```

Comment:
"rederive"

I feel like we could replace each "rederive" with "derive" and it would mean the same thing in the context of a program derived address?

Reply:
like, i get that we're verifying a match from a given pda and "deriving it again" w/the seeds... prolly just me, feels awk, but if it's the way it's spoken of, all good :)